src/SFA.DAS.LevyTransferMatching.Web/Authentication/EmployerAccountAuthorizationHandler.cs

```diff
@@ -0,0 +1,128 @@
+
+        public EmployerAccountAuthorizationHandler(IHttpContextAccessor httpContextAccessor, IAccountUserService accountsService, IConfiguration configuration, ILogger<EmployerAccountAuthorizationHandler> logger)
+        {
+            _httpContextAccessor = httpContextAccessor;
+            _accountsService = accountsService;
+            _configuration = configuration;
+            _logger = logger;
+        }
+        
+        public async Task<bool> IsEmployerAuthorized(AuthorizationHandlerContext context, UserRole minimumAllowedRole)
+        {
+            if (!_httpContextAccessor.HttpContext.Request.RouteValues.ContainsKey(RouteValueKeys.EncodedAccountId))
+            {
+                return false;
+            }
+            var accountIdFromUrl = _httpContextAccessor.HttpContext.Request.RouteValues[RouteValueKeys.EncodedAccountId].ToString().ToUpper();
+            var employerAccountClaim = context.User.FindFirst(c=>c.Type.Equals(ClaimIdentifierConfiguration.Account));
+
+            if(employerAccountClaim?.Value == null)
+                return false;
+
+            Dictionary<string, EmployerUserAccountItem> employerAccounts;
+
+            try
+            {
+                employerAccounts = JsonConvert.DeserializeObject<Dictionary<string, EmployerUserAccountItem>>(employerAccountClaim.Value);
+            }
+            catch (JsonSerializationException e)
+            {
+                _logger.LogError(e, "Could not deserialize employer account claim for user");
+                return false;
+            }
+
+            EmployerUserAccountItem employerIdentifier = null;
+
+            if (employerAccounts != null)
+            {
+                employerIdentifier = employerAccounts.ContainsKey(accountIdFromUrl) 
+                    ? employerAccounts[accountIdFromUrl] : null;
+            }
+
+            if (employerAccounts == null || !employerAccounts.ContainsKey(accountIdFromUrl))
+            {
+                string requiredIdClaim = ClaimIdentifierConfiguration.Id;
+
+                if (_configuration[$"{nameof(Infrastructure.Configuration.FeatureToggles)}:UseGovSignIn"] != null
+                    && _configuration[$"{nameof(Infrastructure.Configuration.FeatureToggles)}:UseGovSignIn"]
+                        .Equals("true", StringComparison.CurrentCultureIgnoreCase))
+                {
+                    requiredIdClaim = ClaimTypes.NameIdentifier;
+                }
+                
+                if (!context.User.HasClaim(c => c.Type.Equals(requiredIdClaim)))
+                    return false;
+                
+                var userClaim = context.User.Claims
+                    .First(c => c.Type.Equals(requiredIdClaim));
+
+                var email = context.User.Claims.FirstOrDefault(c => c.Type.Equals(ClaimTypes.Email))?.Value;
+
+                var userId = userClaim.Value;
+
+                var result = _accountsService.GetUserAccounts(userId, email).Result;
```

Comment:
It looks to me like these parameters are the wrong way around.

Reply:
good spot! have corrected 🕺🏻